packages/core/src/modules/dids/methods/webvh/WebVHDidResolver.ts

```diff
@@ -0,0 +1,38 @@
+import type { AgentContext } from '../../../../agent'
+import type { DidResolver } from '../../domain/DidResolver'
+import type { DidResolutionResult } from '../../types'
+
+import { resolveDID } from 'didwebvh-ts'
+
+import { JsonTransformer } from '../../../../utils/JsonTransformer'
+import { DidDocument } from '../../domain'
+
+export class WebVHDidResolver implements DidResolver {
+  public readonly supportedMethods = ['webvh']
+
+  public readonly allowsCaching = true
+  public readonly allowsLocalDidRecord = true
+
+  // FIXME: Would be nice if we don't have to provide a did resolver instance
+  // private _resolverInstance = new Resolver()
+  // private resolver = didWeb.getResolver()
+
+  public async resolve(agentContext: AgentContext, did: string): Promise<DidResolutionResult> {
+    const result = await resolveDID(did)
```

Comment:
This should probably have a try catch, as it seems the `resolveDID` can throw?

Reply:
thanks, will update!